src/treelearner/linear_tree_learner.cpp

```diff
@@ -211,7 +211,7 @@ void LinearTreeLearner::CalculateLinear(Tree* tree, bool is_refit, const score_t
     } else {
       raw_features = tree->branch_features(i);
     }
-    std::sort(raw_features.begin(), raw_features.end());
+    std::stable_sort(raw_features.begin(), raw_features.end());
```

Comment:
Not sure about this. Refer to https://github.com/microsoft/LightGBM/pull/2154#issuecomment-490716927 and https://github.com/microsoft/LightGBM/pull/1739.
@btrotta Is there a possibility to meet equivalent elements in `raw_features`?
cc @guolinke

Reply:
I think it is not need for stable-sort. I think here is only to get unique feature indices.

Reply:
@guolinke 
Thanks! Reverted in 4358b4a783074fbd520e4882a79d1092ad9875ff.